packages/mdl-menu/simple/foundation.js

```diff
@@ -206,14 +192,110 @@ export default class MDLSimpleMenuFoundation extends MDLFoundation {
     }
   }
 
+  focusOnOpen_() {
+    // First, try focusing the menu.
+    this.adapter_.focusMenu();
+    // If that doesn't work, focus first item instead.
+    if (!this.adapter_.isMenuFocused()) {
+      this.adapter_.focusItemAtIndex(0);
+    }
+  }
+
+  // Handle keys that we want to repeat on hold (tab and arrows).
+  handleKeyboardDown_(evt) {
+    // Do nothing if Alt, Ctrl or Meta are pressed.
+    if (evt.altKey || evt.ctrlKey || evt.metaKey) {
+      return true;
+    }
+
+    const {keyCode, key, shiftKey} = evt;
+    const isTab = key === 'Tab' || keyCode === 9;
+    const isArrowUp = key === 'ArrowUp' || keyCode === 38;
+    const isArrowDown = key === 'ArrowDown' || keyCode === 40;
+
+    const focusedItem = this.adapter_.getFocusedItemIndex();
+    const lastItem = this.adapter_.getNumberOfItems() - 1;
+
+    if (shiftKey && isTab && focusedItem === 0) {
+      this.adapter_.focusItemAtIndex(lastItem);
+      evt.preventDefault();
+      return false;
+    }
+
+    if (!shiftKey && isTab && focusedItem === lastItem) {
+      this.adapter_.focusItemAtIndex(0);
+      evt.preventDefault();
+      return false;
+    }
+
+    if (isArrowUp) {
+      if (focusedItem === 0 || this.adapter_.isMenuFocused()) {
+        this.adapter_.focusItemAtIndex(lastItem);
+      } else {
+        this.adapter_.focusItemAtIndex(focusedItem - 1);
+      }
+    }
+
+    if (isArrowDown) {
+      if (focusedItem === lastItem || this.adapter_.isMenuFocused()) {
+        this.adapter_.focusItemAtIndex(0);
+      } else {
+        this.adapter_.focusItemAtIndex(focusedItem + 1);
+      }
+    }
+
+    return true;
+  }
+
+  // Handle keys that we don't want to repeat on hold (Enter, Space, Escape).
+  handleKeyboardUp_(evt) {
+    // Do nothing if Alt, Ctrl or Meta are pressed.
+    if (evt.altKey || evt.ctrlKey || evt.metaKey) {
+      return true;
+    }
+
+    const {keyCode, key} = evt;
+    const isEnter = key === 'Enter' || keyCode === 13;
+    const isSpace = key === 'Space' || keyCode === 32;
+    const isEscape = key === 'Escape' || keyCode === 27;
+
+    if (isEnter || isSpace) {
+      this.handlePossibleSelected_(evt);
+    }
+
+    if (isEscape) {
+      this.close();
```

Comment:
As discussed offline we need to provide a notification that this cancellation happened, e.g. `notifyCancel()` in the foundation and a `MDLSimpleMenu:cancel` custom event in the component. This would also happen if a users click somewhere outside of the menu.